src/compiler/emitter.ts

```diff
@@ -3991,63 +3999,146 @@ module ts {
-                    // - import declaration is top level and target is value imported by entity name
-                    emitImportDeclaration = !isExternalModule(currentSourceFile) && resolver.isTopLevelValueImportWithEntityName(node);
+            function emitRequire(moduleName: Expression) {
+                if (moduleName.kind === SyntaxKind.StringLiteral) {
+                    write("require(");
+                    emitStart(moduleName);
+                    emitLiteral(<LiteralExpression>moduleName);
+                    emitEnd(moduleName);
+                    emitToken(SyntaxKind.CloseParenToken, moduleName.end);
+                    write(";");
                 }
+                else {
+                    write("require();");
+                }
+            }
 
-                if (emitImportDeclaration) {
-                    if (isExternalModuleImportDeclaration(node) && node.parent.kind === SyntaxKind.SourceFile && compilerOptions.module === ModuleKind.AMD) {
-                        if (node.flags & NodeFlags.Export) {
-                            writeLine();
-                            emitLeadingComments(node);
-                            emitStart(node);
-                            emitModuleMemberName(node);
-                            write(" = ");
-                            emit(node.name);
-                            write(";");
-                            emitEnd(node);
-                            emitTrailingComments(node);
-                        }
-                    }
-                    else {
-                        writeLine();
+            function emitImportAssignment(node: Declaration, moduleName: Expression) {
+                if (!(node.flags & NodeFlags.Export)) write("var ");
+                emitModuleMemberName(<Declaration>node);
+                write(" = ");
+                emitRequire(moduleName);
+            }
+
+            function emitImportDeclaration(node: ImportDeclaration | ImportEqualsDeclaration) {
+                var info = getExternalImportInfo(node);
+                if (info) {
+                    var declarationNode = info.declarationNode;
+                    var namedImports = info.namedImports;
+                    if (compilerOptions.module !== ModuleKind.AMD) {
                         emitLeadingComments(node);
                         emitStart(node);
-                        if (!(node.flags & NodeFlags.Export)) write("var ");
-                        emitModuleMemberName(node);
-                        write(" = ");
-                        if (isInternalModuleImportDeclaration(node)) {
-                            emit(node.moduleReference);
+                        var moduleName = getImportedModuleName(node);
+                        if (declarationNode) {
+                            if (!(declarationNode.flags & NodeFlags.Export)) write("var ");
+                            emitModuleMemberName(declarationNode);

```

Comment:
emitModuleMemberName doesn't (and shouldn't) know whether the surrounding context requires a var keyword to be emitted first. It checks the NodeFlags.Export flag on the node to know whether to rewrite, but that's pretty much it.